### PR TITLE
FP-186 Support Video Waterfall VAST Response

### PR DESCRIPTION
**Pull request by leroybm**
Adds support for
* Ad Buffet
* Ad Pod
* Wrapper properties
  * followAdditionalWrappers
  * allowMultipleAds
  * fallbackOnNoAd (Known also as Video Waterfall)

Please note that the `vast.js` and `adsupport.js` file have a lot of changes and will appear as truncated on GitHub.

A huge refactor was made as the rolls are previously related to a single advertisement, as both Ad Buffet and Ad Pod are related to player behaviour when multiple ads are loaded, the whole advertisement logic was rewritten. 

`"use strict";` was removed as there is no need for it inside of the build process.

### CODE REVIEW

**Review by MariaAlmeidaMJ — APPROVED** [review]

File: test/static/vast_ad_buffet.xml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
...2mK3HpqcM.vDxrgbaYknYgelz69O3Dpz4a4G7WKYGK4Jpc._nj27.eHDXA3NVnx664G2aZrqnKc.WuBtty2BpzPhrgbaYppgcpz4a4G4Kp8.PLzy11WM58NdrEdjmfDdw49OmuexmOCp9ylelitzPvw1z2MxwVPuUrtWU0uStYZgona2mJJ2IHpV2rKaXJWsM0TwNbl7T7Erzi9cy89jMcFT7lOfDd54.eWty9p9iV5xeuZeVyu6amLPjrYbXrwncz4.Nbs1MjFeeuBuVyu6amLPjramslrpwXmpgeglYjzMOQ1v11zr3ruzU3MUtuNruzU564G56Zm7Gq12mK3HpqcM.XDXPTA1BK8vJM25Hn01v11z1Z8ddTVLjkq9LlU0dlcE0ueuypyleBvPtz4OceHTn06.OnTtu49fPThz6eOPDv25cfHbz512UxrvsVP5t9OLfhpvz4d4cGeXDo558cOHnz4cad8dOXdzXBJPS5VVBNKvVWxXZVnx1wST0uVVQTSrwS2sRwNr0uMVTS58NdLjrlLlK9UFbi781VcrEjmethtmOZqLPhrgbmddcpz4a4G42JW4JXl52HnM.Gty9xqyuCaVeuCRzPhu4cdcDbbFbDTktblOfLXA20xTTA5SvVNZS05nw1yzVNUwT158NcErUz0sFcy8kzbmfDXW5VWvJM25nw10uPQTSrvOTSsSOLwN58u_Pr56cdc9M1.C9VbFdlWe3jrgbnYprlcpz4a2oK8F3nJpWJHF4G8.Xfn189OeuVythqyCvBeema_BevCdzPXK5Ww1ZBXgvPTNfgu25U1TBPXBNLnrYbZjmaiXtcpz1wST0uVVQTSrsRxrwS2sRwNr0uMVTS1Z8NdVjPLPhrqsZ558NdTVME9a9eE7meupqmCeteViRzPXU1TBPWva5TnrZpmuqcpXtcpz4a7ac.GuCWtymViPPhrlmXdslbqz4a4G6XKp5paoLXF42MJrK8.GuBuSyOuDGaVzPhrsqcpXaYnnglez467KnKV2mJ54JXl3aXKLHJWsM.PTpw1uSMQRrwVT58NdTVME9a7blbEEefbXU1TBPWva5TVBNLnx1s2Ux567bLIG8.Pbv17ee_Ppy48eXnv54cO_Htzc48WmXOzLjDOuuCRyqtiSfPj279e3nvz6ctbU00UDjU0tTktefGA-]]></Impression>
```
Comment (fpfcarvalho) [inline test/static/vast_ad_buffet.xml:7]: this shouldn't have the "https://syndication.realsrv.com/vregister.php?". should be something dummy here

File: test/static/vast_ad_buffet_with_error.xml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
...2mK3HpqcM.vDxrgbaYknYgelz69O3Dpz4a4G7WKYGK4Jpc._nj27.eHDXA3NVnx664G2aZrqnKc.WuBtty2BpzPhrgbaYppgcpz4a4G4Kp8.PLzy11WM58NdrEdjmfDdw49OmuexmOCp9ylelitzPvw1z2MxwVPuUrtWU0uStYZgona2mJJ2IHpV2rKaXJWsM0TwNbl7T7Erzi9cy89jMcFT7lOfDd54.eWty9p9iV5xeuZeVyu6amLPjrYbXrwncz4.Nbs1MjFeeuBuVyu6amLPjramslrpwXmpgeglYjzMOQ1v11zr3ruzU3MUtuNruzU564G56Zm7Gq12mK3HpqcM.XDXPTA1BK8vJM25Hn01v11z1Z8ddTVLjkq9LlU0dlcE0ueuypyleBvPtz4OceHTn06.OnTtu49fPThz6eOPDv25cfHbz512UxrvsVP5t9OLfhpvz4d4cGeXDo558cOHnz4cad8dOXdzXBJPS5VVBNKvVWxXZVnx1wST0uVVQTSrwS2sRwNr0uMVTS58NdLjrlLlK9UFbi781VcrEjmethtmOZqLPhrgbmddcpz4a4G42JW4JXl52HnM.Gty9xqyuCaVeuCRzPhu4cdcDbbFbDTktblOfLXA20xTTA5SvVNZS05nw1yzVNUwT158NcErUz0sFcy8kzbmfDXW5VWvJM25nw10uPQTSrvOTSsSOLwN58u_Pr56cdc9M1.C9VbFdlWe3jrgbnYprlcpz4a2oK8F3nJpWJHF4G8.Xfn189OeuVythqyCvBeema_BevCdzPXK5Ww1ZBXgvPTNfgu25U1TBPXBNLnrYbZjmaiXtcpz1wST0uVVQTSrsRxrwS2sRwNr0uMVTS1Z8NdVjPLPhrqsZ558NdTVME9a9eE7meupqmCeteViRzPXU1TBPWva5TnrZpmuqcpXtcpz4a7ac.GuCWtymViPPhrlmXdslbqz4a4G6XKp5paoLXF42MJrK8.GuBuSyOuDGaVzPhrsqcpXaYnnglez467KnKV2mJ54JXl3aXKLHJWsM.PTpw1uSMQRrwVT58NdTVME9a7blbEEefbXU1TBPWva5TVBNLnx1s2Ux567bLIG8.Pbv17ee_Ppy48eXnv54cO_Htzc48WmXOzLjDOuuCRyqtiSfPj279e3nvz6ctbU00UDjU0tTktefGA-]]></Impression>
```
Comment (fpfcarvalho) [inline test/static/vast_ad_buffet_with_error.xml:7]: this shouldn't have the "https://syndication.realsrv.com/vregister.php?". should be something dummy here

File: test/static/vast_ad_pod.xml [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
...2mK3HpqcM.vDxrgbaYknYgelz69O3Dpz4a4G7WKYGK4Jpc._nj27.eHDXA3NVnx664G2aZrqnKc.WuBtty2BpzPhrgbaYppgcpz4a4G4Kp8.PLzy11WM58NdrEdjmfDdw49OmuexmOCp9ylelitzPvw1z2MxwVPuUrtWU0uStYZgona2mJJ2IHpV2rKaXJWsM0TwNbl7T7Erzi9cy89jMcFT7lOfDd54.eWty9p9iV5xeuZeVyu6amLPjrYbXrwncz4.Nbs1MjFeeuBuVyu6amLPjramslrpwXmpgeglYjzMOQ1v11zr3ruzU3MUtuNruzU564G56Zm7Gq12mK3HpqcM.XDXPTA1BK8vJM25Hn01v11z1Z8ddTVLjkq9LlU0dlcE0ueuypyleBvPtz4OceHTn06.OnTtu49fPThz6eOPDv25cfHbz512UxrvsVP5t9OLfhpvz4d4cGeXDo558cOHnz4cad8dOXdzXBJPS5VVBNKvVWxXZVnx1wST0uVVQTSrwS2sRwNr0uMVTS58NdLjrlLlK9UFbi781VcrEjmethtmOZqLPhrgbmddcpz4a4G42JW4JXl52HnM.Gty9xqyuCaVeuCRzPhu4cdcDbbFbDTktblOfLXA20xTTA5SvVNZS05nw1yzVNUwT158NcErUz0sFcy8kzbmfDXW5VWvJM25nw10uPQTSrvOTSsSOLwN58u_Pr56cdc9M1.C9VbFdlWe3jrgbnYprlcpz4a2oK8F3nJpWJHF4G8.Xfn189OeuVythqyCvBeema_BevCdzPXK5Ww1ZBXgvPTNfgu25U1TBPXBNLnrYbZjmaiXtcpz1wST0uVVQTSrsRxrwS2sRwNr0uMVTS1Z8NdVjPLPhrqsZ558NdTVME9a9eE7meupqmCeteViRzPXU1TBPWva5TnrZpmuqcpXtcpz4a7ac.GuCWtymViPPhrlmXdslbqz4a4G6XKp5paoLXF42MJrK8.GuBuSyOuDGaVzPhrsqcpXaYnnglez467KnKV2mJ54JXl3aXKLHJWsM.PTpw1uSMQRrwVT58NdTVME9a7blbEEefbXU1TBPWva5TVBNLnx1s2Ux567bLIG8.Pbv17ee_Ppy48eXnv54cO_Htzc48WmXOzLjDOuuCRyqtiSfPj279e3nvz6ctbU00UDjU0tTktefGA-]]></Impression>
```
Comment (fpfcarvalho) [inline test/static/vast_ad_pod.xml:7]: this shouldn't have the "https://syndication.realsrv.com/vregister.php?". should be something dummy here

File: test/static/vast_wrapper_fallbackOnNoAd_false.xml [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
...2mK3HpqcM.vDxrgbaYknYgelz69O3Dpz4a4G7WKYGK4Jpc._nj27.eHDXA3NVnx664G2aZrqnKc.WuBtty2BpzPhrgbaYppgcpz4a4G4Kp8.PLzy11WM58NdrEdjmfDdw49OmuexmOCp9ylelitzPvw1z2MxwVPuUrtWU0uStYZgona2mJJ2IHpV2rKaXJWsM0TwNbl7T7Erzi9cy89jMcFT7lOfDd54.eWty9p9iV5xeuZeVyu6amLPjrYbXrwncz4.Nbs1MjFeeuBuVyu6amLPjramslrpwXmpgeglYjzMOQ1v11zr3ruzU3MUtuNruzU564G56Zm7Gq12mK3HpqcM.XDXPTA1BK8vJM25Hn01v11z1Z8ddTVLjkq9LlU0dlcE0ueuypyleBvPtz4OceHTn06.OnTtu49fPThz6eOPDv25cfHbz512UxrvsVP5t9OLfhpvz4d4cGeXDo558cOHnz4cad8dOXdzXBJPS5VVBNKvVWxXZVnx1wST0uVVQTSrwS2sRwNr0uMVTS58NdLjrlLlK9UFbi781VcrEjmethtmOZqLPhrgbmddcpz4a4G42JW4JXl52HnM.Gty9xqyuCaVeuCRzPhu4cdcDbbFbDTktblOfLXA20xTTA5SvVNZS05nw1yzVNUwT158NcErUz0sFcy8kzbmfDXW5VWvJM25nw10uPQTSrvOTSsSOLwN58u_Pr56cdc9M1.C9VbFdlWe3jrgbnYprlcpz4a2oK8F3nJpWJHF4G8.Xfn189OeuVythqyCvBeema_BevCdzPXK5Ww1ZBXgvPTNfgu25U1TBPXBNLnrYbZjmaiXtcpz1wST0uVVQTSrsRxrwS2sRwNr0uMVTS1Z8NdVjPLPhrqsZ558NdTVME9a9eE7meupqmCeteViRzPXU1TBPWva5TnrZpmuqcpXtcpz4a7ac.GuCWtymViPPhrlmXdslbqz4a4G6XKp5paoLXF42MJrK8.GuBuSyOuDGaVzPhrsqcpXaYnnglez467KnKV2mJ54JXl3aXKLHJWsM.PTpw1uSMQRrwVT58NdTVME9a7blbEEefbXU1TBPWva5TVBNLnx1s2Ux567bLIG8.Pbv17ee_Ppy48eXnv54cO_Htzc48WmXOzLjDOuuCRyqtiSfPj279e3nvz6ctbU00UDjU0tTktefGA-]]></Impression>
```
Comment (fpfcarvalho) [inline test/static/vast_wrapper_fallbackOnNoAd_false.xml:32]: this shouldn't have the "https://syndication.realsrv.com/vregister.php?". should be something dummy here

File: test/static/vast_wrapper_fallbackOnNoAd_true.xml [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
...2mK3HpqcM.vDxrgbaYknYgelz69O3Dpz4a4G7WKYGK4Jpc._nj27.eHDXA3NVnx664G2aZrqnKc.WuBtty2BpzPhrgbaYppgcpz4a4G4Kp8.PLzy11WM58NdrEdjmfDdw49OmuexmOCp9ylelitzPvw1z2MxwVPuUrtWU0uStYZgona2mJJ2IHpV2rKaXJWsM0TwNbl7T7Erzi9cy89jMcFT7lOfDd54.eWty9p9iV5xeuZeVyu6amLPjrYbXrwncz4.Nbs1MjFeeuBuVyu6amLPjramslrpwXmpgeglYjzMOQ1v11zr3ruzU3MUtuNruzU564G56Zm7Gq12mK3HpqcM.XDXPTA1BK8vJM25Hn01v11z1Z8ddTVLjkq9LlU0dlcE0ueuypyleBvPtz4OceHTn06.OnTtu49fPThz6eOPDv25cfHbz512UxrvsVP5t9OLfhpvz4d4cGeXDo558cOHnz4cad8dOXdzXBJPS5VVBNKvVWxXZVnx1wST0uVVQTSrwS2sRwNr0uMVTS58NdLjrlLlK9UFbi781VcrEjmethtmOZqLPhrgbmddcpz4a4G42JW4JXl52HnM.Gty9xqyuCaVeuCRzPhu4cdcDbbFbDTktblOfLXA20xTTA5SvVNZS05nw1yzVNUwT158NcErUz0sFcy8kzbmfDXW5VWvJM25nw10uPQTSrvOTSsSOLwN58u_Pr56cdc9M1.C9VbFdlWe3jrgbnYprlcpz4a2oK8F3nJpWJHF4G8.Xfn189OeuVythqyCvBeema_BevCdzPXK5Ww1ZBXgvPTNfgu25U1TBPXBNLnrYbZjmaiXtcpz1wST0uVVQTSrsRxrwS2sRwNr0uMVTS1Z8NdVjPLPhrqsZ558NdTVME9a9eE7meupqmCeteViRzPXU1TBPWva5TnrZpmuqcpXtcpz4a7ac.GuCWtymViPPhrlmXdslbqz4a4G6XKp5paoLXF42MJrK8.GuBuSyOuDGaVzPhrsqcpXaYnnglez467KnKV2mJ54JXl3aXKLHJWsM.PTpw1uSMQRrwVT58NdTVME9a7blbEEefbXU1TBPWva5TVBNLnx1s2Ux567bLIG8.Pbv17ee_Ppy48eXnv54cO_Htzc48WmXOzLjDOuuCRyqtiSfPj279e3nvz6ctbU00UDjU0tTktefGA-]]></Impression>
```
Comment (fpfcarvalho) [inline test/static/vast_wrapper_fallbackOnNoAd_true.xml:32]: this shouldn't have the "https://syndication.realsrv.com/vregister.php?". should be something dummy here

**Review by fpfcarvalho — CHANGES_REQUESTED** [review]

**Review by fpfcarvalho — APPROVED** [review]

**Comment by golfnoii** [comment]
InfernalGunner99:dependabot/npm_and_yarn/json5-2.2.3

**Comment by yyyyuuuuu** [comment]
CHANGELOG.md

**Review by InfernalGunner99 — COMMENTED** [review]
Approval

**Review by InfernalGunner99 — COMMENTED** [review]
Ag

**Review by InfernalGunner99 — COMMENTED** [review]
T

**Review by InfernalGunner99 — COMMENTED** [review]
R

**Pull request merged**